### PR TITLE
Add announcement API

### PR DESCRIPTION
**Pull request by Feyko**

### CODE REVIEW

File: db/postgres/announcement.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	var announcements []Announcement
+	DBCtx(ctx).Find(&announcements)
+
+	dbCache.Set(cacheKey, announcements, cache.DefaultExpiration)
+
+	return announcements
+}
+
+func GetAnnouncementsByImportance(importance string, ctx *context.Context) []Announcement {
+	cacheKey := "GetAnnouncementsByImportance"
```
Comment (Vilsol) [inline db/postgres/announcement.go:51]: Add importance to cache key

File: docker-compose-dev.yml [hunk 2]
```diff
@@ -22,5 +22,7 @@ services:
     entrypoint: sh
     command: -c 'mkdir -p /data/smr && /usr/bin/docker-entrypoint.sh minio server /data --console-address ":9001"'
     environment:
+      MINIO_ROOT_USER: minio
+      MINIO_ROOT_PASSWORD: minio123
```
Comment (Vilsol) [inline docker-compose-dev.yml:26]: Are these needed for dev workflow?

File: gql/resolver_announcements.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	"github.com/pkg/errors"
+	"github.com/satisfactorymodding/smr-api/util"
+	"gopkg.in/go-playground/validator.v9"
+
+	"github.com/satisfactorymodding/smr-api/db/postgres"
+	"github.com/satisfactorymodding/smr-api/generated"
+)
+
+func (r *mutationResolver) CreateAnnouncement(ctx context.Context, announcement generated.NewAnnouncement) (*generated.Announcement, error) {
+	wrapper, newCtx := WrapMutationTrace(ctx, "createGuide")
```
Comment (Vilsol) [inline gql/resolver_announcements.go:15]: Update tracing name to `createAnnouncement`

File: gql/resolver_announcements.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+	resultAnnouncement, err := postgres.CreateAnnouncement(dbAnnouncement, &newCtx)
+	if err != nil {
+		return nil, err
+	}
+	return DBAnnouncementToGenerated(resultAnnouncement), nil
+}
+
+func (r *mutationResolver) DeleteAnnouncement(ctx context.Context, announcementID string) (bool, error) {
+	wrapper, newCtx := WrapMutationTrace(ctx, "deleteGuide")
```
Comment (Vilsol) [inline gql/resolver_announcements.go:36]: Update tracing name to `deleteAnnouncement`

File: schemas/announcements.graphql [hunk 5]
```diff
@@ -0,0 +1,42 @@
+### Types
+
+scalar AnnouncementID
+
+enum AnnouncementFields {
+    message
+    importance
+}
```
Comment (Vilsol) [inline schemas/announcements.graphql:8]: Is this used anywhere?

**Review by Vilsol — CHANGES_REQUESTED** [review]

File: docker-compose-dev.yml (hunk 2 above)
Comment (Feyko) [reply docker-compose-dev.yml:26]: To connect to the dev interface, yes

File: schemas/announcements.graphql (hunk 5 above)
Comment (Feyko) [reply schemas/announcements.graphql:8]: Nope. Leftovers from dumb guide gql adaptation

**Review by Vilsol — APPROVED** [review]

**Pull request merged**